### PR TITLE
net/lwip/sockets.c : Add Null checking for get_socket

### PR DESCRIPTION
**Pull request by jeongchanKim**
WID:10916217 Return value of a function 'get_socket' is dereferenced at sockets.c:585 without checking, but it is usually checked for this function (23/24).

### CODE REVIEW

File: os/net/lwip/src/api/sockets.c [hunk 1]
```diff
@@ -575,6 +575,12 @@ int lwip_accept(int s, struct sockaddr *addr, socklen_t *addrlen)
 	LWIP_ASSERT("invalid socket index", (newsock >= LWIP_SOCKET_OFFSET) && (newsock < NUM_SOCKETS + LWIP_SOCKET_OFFSET));
 	LWIP_ASSERT("newconn->callback == event_callback", newconn->callback == event_callback);
 	nsock = get_socket(newsock, getpid());
+	if (nsock == NULL) {
+		netconn_delete(newconn);
+		free_socket(nsock, 1);
```
Comment (an4967) [inline os/net/lwip/src/api/sockets.c:580]: As 'nsock' is already NULL, "free_socket(nsock, 1)" would occurs an error.
In 'free_socket', It is using 'nsock' without checking NULL.

Comment (jeongchanKim) [reply os/net/lwip/src/api/sockets.c]: Thanks. I made mistake. I wanted to free `newsock`, so fixed.

**Review by an4967 — APPROVED** [review]
LGTM

**Review by pillip8282 — APPROVED** [review]
lgtm

**Pull request merged**